gguf-py/scripts/gguf-dump.py

```diff
@@ -101,25 +101,200 @@ def dump_metadata_json(reader: GGUFReader, args: argparse.Namespace) -> None:
+
+
+def translate_tensor_name(name):
+    words = name.split(".")
+
+    # Source: https://github.com/ggerganov/ggml/blob/master/docs/gguf.md#standardized-tensor-names
+    abbreviation_dictionary = {
+        'token_embd': 'Token embedding',
+        'pos_embd': 'Position embedding',
+        'output_norm': 'Output normalization',
+        'output': 'Output',
+        'attn_norm': 'Attention normalization',
+        'attn_norm_2': 'Attention normalization',
+        'attn_qkv': 'Attention query-key-value',
+        'attn_q': 'Attention query',
+        'attn_k': 'Attention key',
+        'attn_v': 'Attention value',
+        'attn_output': 'Attention output',
+        'ffn_norm': 'Feed-forward network normalization',
+        'ffn_up': 'Feed-forward network "up"',
+        'ffn_gate': 'Feed-forward network "gate"',
+        'ffn_down': 'Feed-forward network "down"',
+        'ffn_gate_inp': 'Expert-routing layer for the Feed-forward network in Mixture of Expert models',
+        'ffn_gate_exp': 'Feed-forward network "gate" layer per expert in Mixture of Expert models',
+        'ffn_down_exp': 'Feed-forward network "down" layer per expert in Mixture of Expert models',
+        'ffn_up_exp': 'Feed-forward network "up" layer per expert in Mixture of Expert models',
+        'ssm_in': 'State space model input projections',
+        'ssm_conv1d': 'State space model rolling/shift',
+        'ssm_x': 'State space model selective parametrization',
+        'ssm_a': 'State space model state compression',
+        'ssm_d': 'State space model skip connection',
+        'ssm_dt': 'State space model time step',
+        'ssm_out': 'State space model output projection',
+        'blk': 'Block'
+    }
+
+    expanded_words = []
+    for word in words:
+        word_norm = word.strip().lower()
+        if word_norm in abbreviation_dictionary:
+            expanded_words.append(abbreviation_dictionary[word_norm].title())
+        else:
+            expanded_words.append(word.title())
+
+    return ' '.join(expanded_words)
+
+
+def dump_markdown_metadata(reader: GGUFReader, args: argparse.Namespace) -> None:
+    host_endian, file_endian = get_file_host_endian(reader)
+    markdown_content = ""
+    markdown_content += f'# {args.model} - GGUF Internal File Dump\n'
+    markdown_content += f'* Endian: {file_endian} endian\n'
+    markdown_content += '\n'
+    markdown_content += '## Key Value Metadata Store\n'
+    markdown_content += f'There is {len(reader.fields)} key/value pair(s) in this file\n'
+    markdown_content += '\n'
+
+    markdown_content += '| POS | TYPE       | Elements | Key                                    | Value                                                                          |\n'
+    markdown_content += '|-----|------------|----------|----------------------------------------|--------------------------------------------------------------------------------|\n'
```

Comment:
Ideally, the width of the columns could be dynamically calculated.

<details><summary>Otherwise, I'm getting misaligned <code>tokenizer.ggml.tokens</code> and <code>tokenizer.ggml.merges</code> (click to expand)</summary>

```markdown
| POS | TYPE       | Elements | Key                                    | Value                                                                          |
|-----|------------|----------|----------------------------------------|--------------------------------------------------------------------------------|
|   1 | UINT32     |        1 | GGUF.version                           | 3                                                                              |
|   2 | UINT64     |        1 | GGUF.tensor_count                      | 189                                                                            |
|   3 | UINT64     |        1 | GGUF.kv_count                          | 26                                                                             |
|   4 | STRING     |        1 | general.architecture                   | 'jamba'                                                                        |
|   5 | STRING     |        1 | general.name                           | 'jamba-900M-v0.13-KIx2'                                                        |
|   6 | UINT32     |        1 | jamba.block_count                      | 12                                                                             |
|   7 | UINT32     |        1 | jamba.context_length                   | 16384                                                                          |
|   8 | UINT32     |        1 | jamba.embedding_length                 | 1024                                                                           |
|   9 | UINT32     |        1 | jamba.feed_forward_length              | 4096                                                                           |
|  10 | UINT32     |        1 | jamba.attention.head_count             | 32                                                                             |
|  11 | [INT32]    |       12 | jamba.attention.head_count_kv          | [ 0, 8, 0, 0, 0, 8, 0,  ... ]                                                  |
|  12 | UINT32     |        1 | jamba.ssm.conv_kernel                  | 4                                                                              |
|  13 | UINT32     |        1 | jamba.ssm.inner_size                   | 2048                                                                           |
|  14 | UINT32     |        1 | jamba.ssm.state_size                   | 16                                                                             |
|  15 | UINT32     |        1 | jamba.ssm.time_step_rank               | 256                                                                            |
|  16 | FLOAT32    |        1 | jamba.attention.layer_norm_rms_epsilon | 1e-06                                                                          |
|  17 | UINT32     |        1 | jamba.expert_count                     | 8                                                                              |
|  18 | UINT32     |        1 | jamba.expert_used_count                | 2                                                                              |
|  19 | UINT32     |        1 | general.file_type                      | 7                                                                              |
|  20 | STRING     |        1 | tokenizer.ggml.model                   | 'gpt2'                                                                         |
|  21 | STRING     |        1 | tokenizer.ggml.pre                     | 'gpt-2'                                                                        |
|  22 | [STRING]   |    65024 | tokenizer.ggml.tokens                  | [ '[PAD6', '\n\x00\x00\x00\x00', '[PAD6', '\n\x00\x00\x00\x00', '[PAD6',  ... ] |
|  23 | [INT32]    |    65024 | tokenizer.ggml.token_type              | [ 4, 4, 4, 4, 4, 4, 4,  ... ]                                                  |
|  24 | [STRING]   |    64739 | tokenizer.ggml.merges                  | [ 'W ere', '\x05\x00\x00\x00\x00', 'V ict', '\x05\x00\x00\x00\x00', 'T act',  ... ] |
|  25 | UINT32     |        1 | tokenizer.ggml.bos_token_id            | 0                                                                              |
|  26 | UINT32     |        1 | tokenizer.ggml.eos_token_id            | 0                                                                              |
|  27 | UINT32     |        1 | tokenizer.ggml.unknown_token_id        | 0                                                                              |
|  28 | UINT32     |        1 | tokenizer.ggml.padding_token_id        | 0                                                                              |
|  29 | UINT32     |        1 | general.quantization_version           | 2                                                                              |
```

---

</details>

Reply:
should be fixed now. I did it this way because I didn't want to add too much code for it.